react/src/components/AssetDetail/AssetDetail.tsx

```diff
@@ -0,0 +1,172 @@
+import React, { Suspense, useState, useEffect } from 'react';
+import _ from 'lodash';
+import { useAppConfiguration, useFeatureSelection } from '@hazmapper/hooks';
+import { useLocation, useNavigate } from 'react-router-dom';
+import { Asset, FeatureTypeNullable } from '@hazmapper/types';
+import { FeatureIcon } from '@hazmapper/components/FeatureIcon';
+import { Button, LoadingSpinner, SectionMessage } from '@tacc/core-components';
+import styles from './AssetDetail.module.css';
+import { faCameraRetro, faPhotoFilm } from '@fortawesome/free-solid-svg-icons';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+
+type AssetModalProps = {
+  isOpen: boolean;
+  projectId: number;
+  isPublicView: boolean;
+};
+
+const AssetDetail: React.FC<AssetModalProps> = ({
+  isOpen,
+  projectId,
+  isPublicView,
+}) => {
+  const { selectedFeature, selectedFeatureId, setSelectedFeatureId } =
+    useFeatureSelection();
+
+  const navigate = useNavigate();
+  const location = useLocation();
+
+  const closePanel = () => {
+    const params = new URLSearchParams(location.search);
+    params.delete('selectedFeature');
+    navigate(`${location.pathname}?${params.toString()}`);
+  };
+
+  const config = useAppConfiguration();
+  const geoapiUrl = config.geoapiUrl;
+
+  const [selectedFeatureAsset, setSelectedFeatureAsset] = useState<
+    Asset | undefined
+  >(selectedFeature?.assets[0]);
+
+  useEffect(() => {
+    const featureAsset = selectedFeature?.assets[0];
+    setSelectedFeatureAsset(featureAsset);
+  }, [selectedFeature]);
+
+  const featureSource: string | undefined =
+    geoapiUrl + '/assets/' + selectedFeatureAsset?.path;
+
+  const fileType: string | undefined = selectedFeatureAsset?.asset_type;
+
+  const AssetRenderer = React.memo(
+    ({
+      type,
+      source,
+    }: {
+      type: string | undefined;
+      source: string | undefined;
+    }) => {
+      switch (type) {
+        case 'image':
+          return <img src={source} alt="Asset" loading="lazy" width="100%" />;
+        case 'video':
+          return (
+            <video src={source} controls preload="metadata" height="90%" />
+          );
+        case 'point_cloud':
+          /*TODO Add pointcloud */
+          return <div> source={source}</div>;
+        case 'questionnaire':
+          /*TODO Add questionnaire */
+          return <div> source={source}</div>;
+        default:
+          return null;
+      }
+    }
+  );
+
+  return (
+    <div className={styles.root}>
+      <div className={styles.topSection}>
+        <FeatureIcon featureType={fileType as FeatureTypeNullable} />
+        {selectedFeature?.id}
```

Comment:
need to show file name of asset (if there is an asset); see https://github.com/TACC-Cloud/hazmapper/blob/main/angular/src/app/components/asset-detail/asset-detail.component.ts#L58-L63 .  and then fallback is the id